source/common/ssl/context_manager_impl.cc

```diff
@@ -86,7 +86,7 @@ ServerContextPtr ContextManagerImpl::createSslServerContext(
 ServerContext* ContextManagerImpl::findSslServerContext(const std::string& listener_name,
                                                         const std::string& server_name) {
   std::shared_lock<std::shared_timed_mutex> lock(contexts_lock_);
-  if (map_exact_[listener_name][server_name] != nullptr) {
+  if (map_exact_[listener_name].find(server_name) != map_exact_[listener_name].end()) {
```

Comment:
This is functionally correct, but it uses 5 hash table lookups when it only needs to do two.  Should be:

auto& listener_map = map_exact_[listener_name];
auto server_name_it = listener_map.find(server_name);
if (server_name_it != listener_map.end()) {
    return *server_name_it;
}

Same in other places using this map.